### PR TITLE
NIFI-3466 - Addresses ParseCEF Locale related failures and introduce ability to define Locale of  logs

### PR DESCRIPTION
**Pull request by trixpan**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [x] ~~~If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?~~~ 
- [x] ~~~If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?~~~
- [x] ~~~If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?~~~
- [x] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by trixpan** [comment]
@pvillard31 @apiri this should hopefully address the Locale related errors affecting ParseCEF.

**Comment by trixpan** [comment]
also please note there seem to be an unrelated issue affecting another test unit that fails unter pt_BR and fr_FR but passes ja_JP. I suspect it is a comma vs dot number notation issue. 

```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running org.apache.nifi.schemaregistry.processors.TransformersTest
Tests run: 14, Failures: 0, Errors: 3, Skipped: 0, Time elapsed: 0.518 sec <<< FAILURE! - in org.apache.nifi.schemaregistry.processors.TransformersTest
[5] input_csv/decimal_logicalType.txt,input_avro/decimal_logicalType_valid_scale_with_no_default.txt,expected_ouput_csv/decimal_logicalType.txt (testCSVRoundtrip)(org.apache.nifi.schemaregistry.processors.TransformersTest)  Time elapsed: 0.013 sec  <<< ERROR!
java.lang.NumberFormatException: null
	at java.math.BigDecimal.<init>(BigDecimal.java:494)
	at java.math.BigDecimal.<init>(BigDecimal.java:383)
	at java.math.BigDecimal.<init>(BigDecimal.java:806)
	at org.apache.nifi.schemaregistry.processors.CSVUtils.write(CSVUtils.java:161)
	at org.apache.nifi.schemaregistry.processors.TransformersTest.testCSVRoundtrip(TransformersTest.java:149)

[6] input_csv/decimal_logicalType.txt,input_avro/decimal_logicalType_invalid_scale_with_default.txt,expected_ouput_csv/decimal_logicalType_invalid_scale.txt (testCSVRoundtrip)(org.apache.nifi.schemaregistry.processors.TransformersTest)  Time elapsed: 0.004 sec  <<< ERROR!
java.lang.NumberFormatException: null
	at java.math.BigDecimal.<init>(BigDecimal.java:494)
	at java.math.BigDecimal.<init>(BigDecimal.java:383)
	at java.math.BigDecimal.<init>(BigDecimal.java:806)
	at org.apache.nifi.schemaregistry.processors.CSVUtils.write(CSVUtils.java:161)
	at org.apache.nifi.schemaregistry.processors.TransformersTest.testCSVRoundtrip(TransformersTest.java:149)

[7] input_csv/decimal_logicalType_missing_value.txt,input_avro/decimal_logicalType_valid_scale_with_default.txt,expected_ouput_csv/decimal_logicalType_valid_scale_with_default.txt (testCSVRoundtrip)(org.apache.nifi.schemaregistry.processors.TransformersTest)  Time elapsed: 0.005 sec  <<< ERROR!
java.lang.NumberFormatException: null
	at java.math.BigDecimal.<init>(BigDecimal.java:494)
	at java.math.BigDecimal.<init>(BigDecimal.java:383)
	at java.math.BigDecimal.<init>(BigDecimal.java:806)
	at org.apache.nifi.schemaregistry.processors.CSVUtils.write(CSVUtils.java:161)
	at org.apache.nifi.schemaregistry.processors.TransformersTest.testCSVRoundtrip(TransformersTest.java:149)


Results :


Tests in error: 
  TransformersTest.testCSVRoundtrip:149 Â» NumberFormat
  TransformersTest.testCSVRoundtrip:149 Â» NumberFormat
  TransformersTest.testCSVRoundtrip:149 Â» NumberFormat

```

**Comment by trixpan** [comment]
please ignore the last comment regarding the error above, seems to be a rebasing issue.

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ParseCEF.java [hunk 1]
```diff
@@ -146,6 +150,18 @@
...
 
+    public static final PropertyDescriptor DATETIME_REPRESENTATION = new PropertyDescriptor.Builder()
+        .name("DATETIME_REPRESENTATION")
+        .displayName("DateTime Locale")
+        .description("The IETF BCP 47 representation of the Locale to be used when parsing date " +
+                "fields with long or short month names (e.g. may <en-US> vs. mai. <fr-FR>. The default" +
+                "value is generally safe. Only change if having issues parsing CEF messages")
+        .required(true)
+        .addValidator(new ValidateLocale())
+//        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
```
Comment (pvillard31) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ParseCEF.java]: to remove?

**Comment by pvillard31** [comment]
Thanks @trixpan, LGTM, merging to master, thanks for all the efforts you put into locale related build stuff!

**Pull request closed without merging**